pandera/backends/pandas/base.py

```diff
@@ -196,7 +205,18 @@ def drop_invalid_rows(self, check_obj, error_handler: ErrorHandler):
             if isinstance(check_obj.index, pd.MultiIndex):
                 # MultiIndex values are saved on the error as strings so need to be cast back
                 # to their original types
-                index_tuples = err.failure_cases["index"].apply(eval)
+                # fmt: off
+                index_tuples = (
+                    err.failure_cases["index"]
+                    .astype(str)
+                    .apply(
+                        lambda i: eval(i, _MULTIINDEX_HANDLED_TYPES)  # pylint: disable=eval-used
```

Comment:
hmm, I didn't realize when this `eval` snuck into the codebase... are you aware of a more secure way of doing this @rorymcstay ?

Reply:
looks like I missed the `eval` change reviewing this change: https://github.com/unionai-oss/pandera/commit/d43a11b5090de60897f900479fae3dd68b016415

We don't need to address in this PR, but if you can think of a non-eval approach to this in a separate PR that would be awesome!